src/concurrent.rs

```diff
@@ -72,6 +72,16 @@ impl ShareableFile {
         }
     }
 
+    /// Constructs a new `ShareableFile` from an stdin handle and takes ownership of it.
+    pub fn from_stdin(stdin: io::Stdin) -> ShareableFile {
+        use std::os::unix::io::AsRawFd;
+        ShareableFile {
+            fd: stdin.as_raw_fd(),
```

Comment:
Ehhhh, technically you're trying to steal it when it's not yours. Is it worth dup'ing it into a File so you don't have to propagate so much special case?

Reply:
Uh, I didn't think about nor notice I could have created a `File` from the standard descriptors. Requires an `unsafe` block... but oh well, it seems much cleaner than what I did.

PTAL; this second commit is essentially redone now.